pkg/api/v1alpha1/cloudstackmachineconfig.go

```diff
@@ -85,6 +85,15 @@ func GetCloudStackMachineConfigs(fileName string) (map[string]*CloudStackMachine
 	return configs, nil
 }
 
+func setCloudStackMachineConfigDefaults(machineConfig *CloudStackMachineConfig) {
+	if len(machineConfig.Spec.Users) <= 0 {
```

Comment:
Setting the ssh key to empty string means the user can never access nodes without the ssh keys right? Instead of these defaults, I vote we add validation for the controller where the user should provide a ssh key, atleast till we start generating ssh keys for the controller.

Reply:
Do we want to do this for all the other providers as well? If so, maybe it would be good idea to make those changes in this PR or in another PR all at once.

Reply:
I believe we set the set defaults in the CLI to avoid a nil pointer when generating the keys.

But yeah for the controller, I agree that it makes more sense to validate rather than defaulting to empty. Will add that validation & I'll check the other providers to do the same if it doesn't already

Reply:
I do have the code changes to fix it for other providers. I picked up that action item when I noticed this default issue for baremetal.

Reply:
Ok awesome. I can close this PR out then & let you do it for CloudStack too in that PR